### ISSUE TITLE
Eval/Training curve mutates during training

### ISSUE THREAD

**Issue by 1021331023** [SAGNIKMJR/few-shot-rir#2]
Hi! Thanks for your awesome work.

I recently tried to run the code and found that the eval/training curve mutates in about 140 epochs during training.
Here are the results.
![image](https://github.com/SAGNIKMJR/few-shot-rir/assets/55610694/636aafa0-f203-4d76-b389-e16bb10a1a56)

Could you provide some insights on this phenomenon? 

Appreciate.

**Comment by SAGNIKMJR** [comment]
Hi,

Thanks for your interest in the work!

I also noticed this behavior during training. I am not entirely sure but it's possible that the model hits some local optimum through the drop, with respect to to unseen environments (see unseen eval curve). The combination of Adam and our learning rate choice could be responsible for it. It's also possible that if you train for more epochs than I did, the model will improve further. If you find anything like that through your experiments, let me know and we can think of updating the repo with these results/observations.
